### PR TITLE
Add subtitle slot to the Primer::Alpha::Overlay::Header and Primer::Alpha::Dialog::Header

### PR DESCRIPTION
**Pull request by kendallgassner**
### What are you trying to accomplish?

I would like to add a link inside of a subtitle. In order to do this I need the subtitle field to be a slot.

#### List the issues that this change affects.

Closes # (type the GitHub issue number after #)

#### Risk Assessment
  

- [ ] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **Fixes axe scan violation** - This change fixes an existing [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation.
- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.
- **New axe violation** - This change introduces a new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation. Please describe why the violation cannot be resolved below.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Review by camertron — COMMENTED** [review]
Would you mind adding the same slot to the `Dialog` header so the APIs are consistent?

File: test/components/primer/alpha/dialog_test.rb [hunk 1]
```diff
@@ -91,6 +91,12 @@ def test_renders_large_header
         end
       end
 
+      def test_renders_header_subtitle
```
Comment (camertron) [inline test/components/primer/alpha/dialog_test.rb:94]: Are we missing an assertion in this test?

**Review by camertron — CHANGES_REQUESTED** [review]

**Review by camertron — APPROVED** [review]

**Comment by kendallgassner** [comment]
@camertron can you merge this 🙏

File: app/components/primer/alpha/dialog/header.rb [hunk 2]
```diff
@@ -28,6 +28,20 @@ class Header < Primer::Component
           Primer::BaseComponent.new(**system_arguments)
         }
 
+                # Optional subtitle slot for adding a subtitle to the header.
+        #
```
Comment (camertron) [inline app/components/primer/alpha/dialog/header.rb:32]:
```suggestion
        # Optional subtitle slot for adding a subtitle to the header.
        #
```

**Pull request merged**